resource-timing/buffer-full-add-after-full-event.html

```diff
@@ -53,12 +53,20 @@
 };
 
 let testThatEntryWasAdded = () => {
+    let tries = 1;
+    let maxTries = 5;
     return new Promise((resolve, reject) => {
         let waitForIt = function() {
             if (performance.getEntriesByType("resource").length) {
                 resolve();
             } else {
-                reject("After buffer full, entry never added to primary");
+                if (tries < maxTries) {
+                    tries++;
+                    step_timeout(waitForIt, 0);
+                } else {
+                    reject("After buffer full (and " + tries +
+                           " task ticks), entry never added to primary");
+                }
             }
         }
         step_timeout(waitForIt, 0);
```

Comment:
This is [not the only place](https://cs.chromium.org/search/?q=waitForNextTask&sq=package:chromium&type=cs) in those tests where we wait for the next task, which may not be a good enough guaranty.

It might be better to rename `waitForNextTask` to something like `waitTillConditionIsMet`, pass it a function that returns a boolean, and move the step_timeout calls and retries there. Does that make sense?

Reply:
I do think that this makes sense -- thank you for the suggestion! Just to make sure that I understood your suggestion completely, here is how I imagine the use of that function:

```
await waitUntilConditionIsMet(() => {
            if (performance.getEntriesByType("resource").length) {
                return true;
            } else {
                if (tries < maxTries) {
                    tries++;
                    return false;
                } else {
                    return true;
                }
            }
        }).then( (resolve, reject) => {
                if (performance.getEntriesByType("resource").length) {
                    resolve();
                } else {
                    reject();
                }
        });
```

Semantically, UntilConditionIsMet will not resolve until function in the parameter returns true.

I actually like the way that looks. 

NB: That was written in a web comment field so it may not be syntactically correct :-)